### PR TITLE
Arm utils 2

### PR DESCRIPTION
**Pull request by kz0817**
These patches extract similar lines in ArmBase and HatoholArmPluginGate to a new class ArmUtils.

### CODE REVIEW

File: server/src/HatoholArmPluginGate.cc [hunk 1]
```diff
@@ -517,70 +457,15 @@ void HatoholArmPluginGate::setPluginAvailabelTrigger(const HatoholArmPluginWatch
 void HatoholArmPluginGate::setPluginConnectStatus(const HatoholArmPluginWatchType &type,
 					     const HatoholArmPluginErrorCode &errorCode)
 {
-	TriggerInfoList triggerInfoList;
-	EventInfoList eventInfoList;
-	TriggerStatusType istatusType;
-
+	TriggerStatusType istatus;
```
Comment (ashie) [inline server/src/HatoholArmPluginGate.cc]: I think the prefix "i" isn't needed.

Comment (kz0817) [inline server/src/HatoholArmPluginGate.cc]: I fixed it. Please resume reviewing.

File: server/src/ArmUtils.cc [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+		if (armTrigger.status != TRIGGER_STATUS_UNKNOWN)
+			obj->createEvent(armTrigger, arg.eventInfoList);
+
+		updateLowerTriggers(arg);
+		updateUpperTriggers(arg);
+	}
+
+	void updateLowerTriggers(UpdateTriggerArg &arg)
+	{
+		for (size_t i = arg.triggerIdx + 1; i < numArmTriggers; i++) {
```
Comment (ashie) [inline server/src/ArmUtils.cc]: The range of the "i" seems same with updateUpperTriggers().
Is it correct?

I think the correct code for it is:
```c++
for (size_t i = 0; i < arg.triggerIdx; i++) {
```

Comment (kz0817) [inline server/src/ArmUtils.cc]: Thank you for the comment. I fixed it and added a unit test.

**Comment by ashie** [comment]
Confirmed:

* I got a same result with the previous implementation on stalling a Zabbix server (both arm & hap).
* Also I got a same result on reviving them.

**Comment by ashie** [comment]
BTW I found a strange behaviour while testing this modification.
When I filter triggers by the hostname of the self-monitiring, 2 set of self-monitoring triggers are shown.

It's not a bug in this modification.
I'll report it as another issue.

**Comment by ashie** [comment]
> I'll report it as another issue.

#1046

**Pull request merged**